sdk/eventhubs/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorAsyncClient.java

```diff
@@ -0,0 +1,181 @@
+
+    private static final long INTERVAL_IN_SECONDS = 10; // run every 10 seconds
+    private static final long INITIAL_DELAY = 0; // start immediately
+    private final ClientLogger logger = new ClientLogger(EventProcessorAsyncClient.class);
+
+    private final EventHubAsyncClient eventHubAsyncClient;
+    private final String consumerGroupName;
+    private final EventPosition initialEventPosition;
+    private final PartitionProcessorFactory partitionProcessorFactory;
+    private final PartitionManager partitionManager;
+    private final Map<String, EventHubConsumer> partitionConsumers = new ConcurrentHashMap<>();
+    private final String instanceId;
+    private final String eventHubName;
+    private Disposable runner;
+    private Scheduler scheduler;
+    private AtomicBoolean started = new AtomicBoolean(false);
+
+    /**
+     * Package-private constructor. Use {@link EventHubClientBuilder} to create an instance.
+     *
+     * @param eventHubAsyncClient The {@link EventHubAsyncClient}.
+     * @param consumerGroupName The consumer group name used in this event processor to consumer events.
+     * @param partitionProcessorFactory The factory to create new partition processors.
+     * @param initialEventPosition Initial event position to start consuming events.
+     * @param partitionManager The partition manager.
+     * @param eventHubName The event hub name.
+     */
+    EventProcessorAsyncClient(EventHubAsyncClient eventHubAsyncClient, String consumerGroupName,
+        PartitionProcessorFactory partitionProcessorFactory, EventPosition initialEventPosition,
+        PartitionManager partitionManager,
+        String eventHubName) {
+        this.eventHubAsyncClient = Objects
+            .requireNonNull(eventHubAsyncClient, "eventHubAsyncClient cannot be null");
+        this.consumerGroupName = Objects
+            .requireNonNull(consumerGroupName, "consumerGroupname cannot be null");
+        this.partitionProcessorFactory = Objects
+            .requireNonNull(partitionProcessorFactory, "partitionProcessorFactory cannot be null");
+        this.partitionManager = Objects
+            .requireNonNull(partitionManager, "partitionManager cannot be null");
+        this.initialEventPosition = Objects
+            .requireNonNull(initialEventPosition, "initialEventPosition cannot be null");
+        this.eventHubName = Objects
+            .requireNonNull(eventHubName, "eventHubName cannot be null");
+        this.instanceId = UUID.randomUUID().toString();
+        logger.info("The instance ID for this event processors is {}", this.instanceId);
+    }
+
+    /**
+     * Starts processing of events for all partitions of the Event Hub that this event processor can own, assigning a
+     * dedicated {@link PartitionProcessor} to each partition. If there are other Event Processors active for the same
+     * consumer group on the Event Hub, responsibility for partitions will be shared between them.
+     * <p>
```

Comment:
IntelliJ added this but add the associated `</p>` tag. It was commented in previous PRs of mine to close my HTML tags.